content/blog/20231114_fluentd-v1.16.2-v1.16.3-have-been-released.md

```diff
@@ -0,0 +1,52 @@
+# Fluentd v1.16.3 and v1.16.2 have been released
+
+Hi users!
+
+We have released v1.16.3 ([changelog](https://github.com/fluent/fluentd/blob/v1.16/CHANGELOG.md#release-v1163---20231114)) on 2023-11-14.
+We also describe the content of v1.16.2 ([changelog](https://github.com/fluent/fluentd/blob/v1.16/CHANGELOG.md#release-v1162---20230714)) on 2023-07-14 here.
+
+In these releases, `in_tails` issues that have been reported for years are fixed!
+And other some bug fixes too.
+
+## Bug Fixes
+
+### `in_tail`: Fix `in_tail` wrongly stopping tailing the current target file and causing handle leaks
+
+This issue has been reported for years.
+It was looking very serious, but the cause remained unclear until recently.
+
+At last, we found different causes when the [follow_inodes option](https://docs.fluentd.org/input/tail#follow_inodes) is `true` and when it is `false` (default).
+We fixed the problem of `follow_inodes true` case in v1.16.2, and fixed the problem of `follow_inodes false` case in v1.16.3.
+
+Especially, in `follow_inodes false` case, the trouble was mainly caused by [rotate_wait option](https://docs.fluentd.org/input/tail#rotate_wait).
+In past versions, please avoid using large value for this option.
+In particular, if you make it greater than [refresh_interval option](https://docs.fluentd.org/input/tail#refresh_interval), it will surely cause this issue.
+
+In Fluentd v1.15.1 (td-agent v4.4.0) or later, the following warning log is recorded when this issue occurs.
+
+```
+Skip update_watcher because watcher has been already updated by other inotify event
+```
+
+If this issue occurs, please restart Fluentd, and please update Fluentd to the latest version.
+fluent-package v5.0.2 will be released in this month and it contains these fixes.
+
+See the following PRs for more information.
+
+* `follow_inodes true` case: [#4208](https://github.com/fluent/fluentd/pull/4208)
+* `follow_inodes false` case: [#4327](https://github.com/fluent/fluentd/pull/4327)
+
+### Others
+
+* `in_tail`: Fix possible log duplication of [follow_inodes option](https://docs.fluentd.org/input/tail#follow_inodes). [#4237](https://github.com/fluent/fluentd/pull/4237), [#4239](https://github.com/fluent/fluentd/pull/4239)
+* `in_forward`: Fix corrupted data possibly breaking other data. [#4178](https://github.com/fluent/fluentd/pull/4178)
+* [ignore_same_log_interval option](https://docs.fluentd.org/deployment/system-config#ignore_same_log_interval): Fix cache size growing unlimitedly. [#4229](https://github.com/fluent/fluentd/pull/4229)
```

Comment:
```suggestion
* [ignore\_same\_log\_interval option](https://docs.fluentd.org/deployment/system-config#ignore_same_log_interval): Fix cache size growing unlimitedly. [#4229](https://github.com/fluent/fluentd/pull/4229)
```

It seems that it requires to be escaped here.

Reply:
![Uploading image.png…]()

Reply:
Thanks! Sorry for overlooking it. Fixed.